test/spawn.jl

```diff
@@ -620,7 +620,9 @@ end
 @test reduce(&, [`$echocmd abc`, `$echocmd def`, `$echocmd hij`]) == `$echocmd abc` & `$echocmd def` & `$echocmd hij`
 
 # readlines(::Cmd), accidentally broken in #20203
-@test sort(readlines(`$lscmd -A`)) == sort(readdir())
+let str = "foo\nbar"
+    @test readlines(`$echocmd $str`) == collect(eachline(IOBuffer(str)))
```

Comment:
`split(str)` seems to be slightly easier to parse than `collect(eachline(IOBuffer(str)))`?
```suggestion
    @test readlines(`$echocmd $str`) == split(str)
```